app/routers/event.py

```diff
@@ -70,65 +77,129 @@ async def create_event_api(event: EventModel, session=Depends(get_db)):
     return {'success': True}
 
 
-@router.get("/edit", include_in_schema=False)
-@router.get("/edit")
+@router.get('/edit', include_in_schema=False)
+@router.get('/edit')
 async def eventedit(request: Request) -> Response:
-    return templates.TemplateResponse("event/eventedit.html",
-                                      {"request": request})
+    return templates.TemplateResponse('event/eventedit.html', {
+        'request': request,
+        'privacy': PrivacyKinds
+    })
 
 
-@router.post("/edit", include_in_schema=False)
-async def create_new_event(request: Request,
-                           session=Depends(get_db)) -> Response:
+@router.post('/edit', include_in_schema=False)
+async def create_new_event(
+    request: Request, session=Depends(get_db)
+) -> Response:
     data = await request.form()
     title = data['title']
     content = data['description']
-    start = dt.strptime(data['start_date'] + ' ' +
-                        data['start_time'], TIME_FORMAT)
-    end = dt.strptime(data['end_date'] + ' ' + data['end_time'],
-                      TIME_FORMAT)
+    start = dt.strptime(data['start_date'] + ' ' + data['start_time'],
+                        TIME_FORMAT)
+    end = dt.strptime(data['end_date'] + ' ' + data['end_time'], TIME_FORMAT)
     owner_id = get_current_user(session).id
     availability = data.get('availability', 'True') == 'True'
     location = data['location']
     vc_link = data['vc_link']
     category_id = data.get('category_id')
+    privacy = data['privacy']
+    privacy_kinds = [kind.name for kind in PrivacyKinds]
+    if privacy not in privacy_kinds:
+        privacy = PrivacyKinds.Public.name
 
     invited_emails = get_invited_emails(data['invited'])
-    uninvited_contacts = get_uninvited_regular_emails(session, owner_id,
-                                                      title, invited_emails)
+    uninvited_contacts = get_uninvited_regular_emails(session, owner_id, title,
+                                                      invited_emails)
 
     if vc_link is not None:
         raise_if_zoom_link_invalid(vc_link)
 
-    event = create_event(db=session, title=title, start=start, end=end,
-                         owner_id=owner_id, content=content,
-                         location=location, vc_link=vc_link,
+    event = create_event(db=session,
+                         title=title,
+                         start=start,
+                         end=end,
+                         owner_id=owner_id,
+                         privacy=privacy,
+                         content=content,
+                         location=location,
                          invitees=invited_emails,
                          category_id=category_id,
                          availability=availability)
```

Comment:
I don't know why this happened...
Almost for all of your comments are stuff I didn't actually change . It's from the prehooks that automatically formatted everything 😭

Reply:
Changed it back to the way it was.

Reply:
So, I know you said that it's preferable to not repeat things but as the function itself has some deafult values that sometimes are necessary or not I changed the formatting but did leave the explicit attributing of each variable to the parameters as I saw that otherwise this causes problems and misunderstanding in tests and so on.
It's less pretty yes, but leaves less room for mistakes and doesn't require to change the create_event function parameters order (especially parameters that are not related to my feature which could confuse or mess up other things)